drivers/gles3/rasterizer_canvas_gles3.cpp

```diff
@@ -249,6 +249,66 @@ void RasterizerCanvasGLES3::_set_texture_rect_mode(bool p_enable) {
 
 void RasterizerCanvasGLES3::_draw_polygon(int p_vertex_count, const int *p_indices, const Vector2 *p_vertices, const Vector2 *p_uvs, const Color *p_colors, const RID &p_texture, bool p_singlecolor) {
 
+	int version = 0;
+	int color_ofs = 0;
+	int uv_ofs = 0;
+	int stride = 2;
+
+	if (p_colors) { //color
+		version |= 1;
+		color_ofs = stride;
+		stride += 4;
+	}
+
+	if (p_uvs) { //uv
+		version |= 2;
+		uv_ofs = stride;
+		stride += 2;
+	}
+
+	float b[(2 + 2 + 4) * p_vertex_count];
```

Comment:
Both options result in the same memory layout, there is no speed difference between the two when copying to the GPU. The end result is the same. PoolRealArray is just the "godot" way.